### ISSUE TITLE
Algorithm problem

### ISSUE THREAD

**Issue by wangzeyu135798** [devnkong/FLAG#1]
Hi:
   After reading your paper and code, I have a question.
   In ogbn_protiens/attacks.py, function flag, I think it's the core of this algorithm. While In this function, you first calculate a loss under data and perturb, in next args.m -1 times, calculate args.m -1 times loss and accumulate gradients of perturb. In last, the total loss backwards. In your code, accumulating several times gradients of perturb while updating model parameters once. It seems that this doesn't match your paper!  
    In your paper algotithm1, from line 6 to 8, the adversarial loop run one time, computing both the gradient for perturb and parameter simultaneously.

**Comment by devnkong** [comment]
Hello, thanks for your interest in our paper!

The gradients will be accumulated for model parameters and perturbations M times in our algorithm. To way to realize it is to loss.backward() M times, but we only do gradient ascent for perturbations in the loop, without optimizing the model parameters. After we go outside the loop we optmize model parameter once. These match both our Algorithm 1 in the paper and our code. 

Hope this makes sense!

**Comment by wangzeyu135798** [comment]
Hi:
&nbsp; &nbsp;I want to learn which conference or magazine do you publish this paper on ?&nbsp;


------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "devnkong/FLAG"                                                                                    <notifications@github.com&gt;;
发送时间:&nbsp;2020年11月16日(星期一) 上午8:45
收件人:&nbsp;"devnkong/FLAG"<FLAG@noreply.github.com&gt;;
抄送:&nbsp;"天津工业大学王泽宇"<1051533398@qq.com&gt;;"Author"<author@noreply.github.com&gt;;
主题:&nbsp;Re: [devnkong/FLAG] Algorithm problem (#1)





 
Hello, thanks for your interest in our paper!
 
The gradients will be accumulated for model parameters and perturbations M times in our algorithm. To way to realize it is to loss.backward() M times, but we only do gradient ascent for perturbations in the loop, without optimizing the model parameters. After we go outside the loop we optmize model parameter once. These match both our Algorithm 1 in the paper and our code.
 
Hope this makes sense!
